gdx-pay-android-googlebilling/src/com/badlogic/gdx/pay/android/googlebilling/PurchaseManagerGoogleBilling.java

```diff
@@ -106,76 +106,79 @@ private String getGlobalSkuTypeFromConfig() {
-                                    informationMap.put(skuDetails.getSku(), convertSkuDetailsToInformation
-                                            (skuDetails));
-                                    skuDetailsMap.put(skuDetails.getSku(), skuDetails);
-                                }
-                            } else {
-                                Gdx.app.log(TAG, "skuDetailsList is null");
+                            for (ProductDetails productDetails : productDetailsList) {
+                                informationMap.put(productDetails.getProductId(), convertSkuDetailsToInformation
+                                        (productDetails));
+                                productDetailsMap.put(productDetails.getProductId(), productDetails);
                             }
+
                             setInstalledAndNotifyObserver();
                         }
                     }
-                });
+                }
+        );
     }
 
     private String mapOfferType(OfferType type) {
         switch (type) {
             case CONSUMABLE:
             case ENTITLEMENT:
-                return BillingClient.SkuType.INAPP;
+                return BillingClient.ProductType.INAPP;
             case SUBSCRIPTION:
-                return BillingClient.SkuType.SUBS;
+                return BillingClient.ProductType.SUBS;
         }
         throw new IllegalStateException("Unsupported OfferType: " + type);
     }
 
-    private Information convertSkuDetailsToInformation(SkuDetails skuDetails) {
-        String priceString = skuDetails.getPrice();
+    private Information convertSkuDetailsToInformation(ProductDetails productDetails) {
+        String priceString = productDetails.getOneTimePurchaseOfferDetails().getFormattedPrice();
         return Information.newBuilder()
-                .localName(skuDetails.getTitle())
-                .freeTrialPeriod(convertToFreeTrialPeriod(skuDetails.getFreeTrialPeriod()))
-                .localDescription(skuDetails.getDescription())
+                .localName(productDetails.getTitle())
+                //TODO Removed - Dont know if this is applied in some other form
+                //.freeTrialPeriod(convertToFreeTrialPeriod(productDetails.getFreeTrialPeriod()))
+                .localDescription(productDetails.getDescription())
                 .localPricing(priceString)
-                .priceCurrencyCode(skuDetails.getPriceCurrencyCode())
-                .priceInCents((int) (skuDetails.getPriceAmountMicros() / 10_000))
-                .priceAsDouble(skuDetails.getPriceAmountMicros() / 1_000_000.0)
+                .priceCurrencyCode(productDetails.getOneTimePurchaseOfferDetails().getPriceCurrencyCode())
+                .priceInCents((int) (productDetails.getOneTimePurchaseOfferDetails().getPriceAmountMicros() / 10_000))
+                .priceAsDouble(productDetails.getOneTimePurchaseOfferDetails().getPriceAmountMicros() / 1_000_000.0)
                 .build();
```

Comment:
getOneTimePurchaesOfferDetails() is incorrect for subscriptions; productDetails.getSubscriptionOfferDetails() should be used, to be defined how to do it further